Project 5 For John Beverly

```diff
@@ -51,3 +51,441 @@ You may find the following resources useful:
+
+] ;
+
+sh:property [    
+
+
+sh:path ex:hasTarget ;    
+
+sh:class ex:Node  
+
+] ;
+
+sh:property [    
+
+sh:path ex:hasValue ;    
+
+sh:datatype xsd:boolean ;    
+
+sh:in ( true )  ] ;
+
+sh:property [    
+
+sh:path ex:hasType ;    
+
+sh:in ( "asymmetric" )  
+
+] .
+
+
+9.	Constraint for PropertyIn for Asymmetrical or Irreflexive 
+a.	Our constraint checks for when a graph has a propertyIn that is both asymmetric and irreflexive. 
+
+b.	An example of a time when you could use this constraint would be when checking a flow chart type graph (I think they are called directed acyclic graphs). Such that in a flow chart there could be many outgoing relations with only one incoming relation. This means the ex:asymmetricalPropertySHape constraint could ensure that each arrow of the graph in directed in a consistent way. While is ex:IrreeflexivePropertyShape constraint on the other hand, could be used to ensure that there are no cycles in the graph by prohibiting self-reference. By combing these constraints with sh:or property type you can check the validity of a flow chart. 
+
+c.	[(a -> b) ^ ~(b -> a)] v [~(a -> a)]
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:asymmetricalPropertyShape
+
+a sh:PropertyShape ;
+
+sh:path ex:asymmetricalProperty ;
+
+sh:propertyConstraint [
+
+sh:asymmetrical true ;
+
+] .
+
+ex:irreflexivePropertyShape
+
+a sh:PropertyShape ;
+
+sh:path ex:irreflexiveProperty ;
+
+sh:propertyConstraint [
+
+sh:irreflexive true ;
+
+] .
+
+ex:MyClassShape
+
+a sh:NodeShape ;
+
+sh:property [
+
+sh:or ( ex:asymmetricalPropertyShape ex:irreflexivePropertyShape ) ;
+
+] .
+
+
+
+10.	Constraint for hasProperty
+a.	An example of when this constraint might be used is in a real estate application that manages multiple properties. Such that the three associated properties rdf:value (the monetary value of a given property), ex:b (the number of bedrooms on a given property) and ex:c (the number of bathrooms on a given property). The constraint can be applied to ensure that any of the data associated with ex:property class meets these requirements, preventing incomplete or incorrect data being entered into the system. 
+
+b.	Going to be perfectly honest. I am not sure how to translate this into symbolic logic. 
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:hasProperty
+
+a sh:NodeShape ;
+
+sh:targetClass ex:property ;
+
+sh:property [    
+
+sh:path rdf:value ;        
+
+sh:minCount 1 ;        
+
+sh:maxCount 8 ;      
+
+] ;
+  
+
+sh:property [    
+
+sh:path ex:b ;    
+
+sh:minCount 1 ;    
+
+sh:maxCount 8 ;  
+
+] ;
+
+sh:property [    
+
+sh:path ex:c ;    
+
+sh:minCount 1 ;    
+
+sh:maxCount 8 ;  
```

Comment:
What does "sh:maxCount 8" mean? Can't it be "sh:maxCount 7"? I'm just wondering how a particular number is specified here.

Reply:
I truly just chose some numbers lol as these are meant to be examples